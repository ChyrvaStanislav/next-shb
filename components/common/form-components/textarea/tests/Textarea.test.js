import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Formik } from 'formik';
import Textarea from '../index';

const setFieldValue = jest.fn();
const onBlur = jest.fn();
const onChange = jest.fn();

const event = {
  target: {
    value: 'some value'
  }
};

const Component = props => (
  <Formik>
    <Textarea
      field={{
        name: 'name',
        value: '',
        onBlur,
        onChange,
      }}
      form={{
        touched: {},
        errors: {},
        setFieldValue,
      }}
      type="text"
      placeholder="Some placeholder"
      formId="formId"
      classNameWrapper="someClass"
      classNameTextarea="anotherClass"
      label="Some Label"
      required={true}
      {...props}
    />
  </Formik>
);

describe('Textarea renders', () => {
  it('should Textarea renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should Textarea renders with counter', () => {
    const { asFragment } = render(<Component isCounter={true} maxLength={125} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Events works', () => {
  it('should onBlur works', () => {
    const { getByTestId } = render(<Component />);
    const textarea = getByTestId('textarea');
    fireEvent.blur(textarea, event);

    expect(onBlur).toBeCalled();
    expect(setFieldValue).toBeCalled();
    expect(textarea.value).toBe('some value');
  });

  it('should onChange works', () => {
    const { getByTestId } = render(<Component />);
    const textarea = getByTestId('textarea');
    fireEvent.change(textarea, event);

    expect(onChange).toBeCalled();
  });
});

describe('Error display', () => {
  it('should Error message displays', () => {
    const { getByText } = render(<Component form={{
      touched: {
        name: true,
      },
      errors: {
        name: 'Some error'
      },
    }}
    />);

    const errorIcon = getByText('Some error');
    expect(errorIcon).toBeInTheDocument();
  });
});
