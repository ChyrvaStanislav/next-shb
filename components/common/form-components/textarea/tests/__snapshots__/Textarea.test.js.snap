// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea renders should Textarea renders 1`] = `
<DocumentFragment>
  <section
    class="textareaContainer"
  >
    <label
      class="small_filters label required"
      for="formId-name"
      other=""
    >
      Some Label
    </label>
    <div
      class="textareaWrapper someClass"
    >
      <textarea
        autocomplete="off"
        class="textarea anotherClass"
        data-testid="textarea"
        id="formId-name"
        name="name"
        placeholder="Some placeholder"
        type="text"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`Textarea renders should Textarea renders with counter 1`] = `
<DocumentFragment>
  <section
    class="textareaContainer"
  >
    <label
      class="small_filters label required"
      for="formId-name"
      other=""
    >
      Some Label
    </label>
    <div
      class="textareaWrapper someClass"
    >
      <textarea
        autocomplete="off"
        class="textarea anotherClass"
        data-testid="textarea"
        id="formId-name"
        maxlength="125"
        name="name"
        placeholder="Some placeholder"
        type="text"
      />
      <span
        class="counter"
      >
        0/125
      </span>
    </div>
  </section>
</DocumentFragment>
`;
